From c422614cc54b667916591c6b56645680ab7c5c88 Mon Sep 17 00:00:00 2001
From: Zheng Yejian <zhengyejian1@huawei.com>
Date: Tue, 23 Aug 2022 15:58:20 +0000
Subject: [PATCH 028/261] [Huawei] livepatch/core: Fix warning
 C_RULE_ID_SINGLE_BRANCH_IF_AND_LOOP_BRACKET

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187321

--------------------------------

Signed-off-by: Zheng Yejian <zhengyejian1@huawei.com>
---
 kernel/livepatch/core.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kernel/livepatch/core.c b/kernel/livepatch/core.c
index ec0921ed1e2f..1a3c0e4e300a 100644
--- a/kernel/livepatch/core.c
+++ b/kernel/livepatch/core.c
@@ -1054,10 +1054,9 @@ static int klp_init_object_loaded(struct klp_patch *patch,
 		 * feature 'function descriptor'), otherwise size found by
 		 * 'kallsyms_lookup_size_offset' may be abnormal.
 		 */
-		if (func->old_name[0] !=  '.') {
+		if (func->old_name[0] !=  '.')
 			pr_warn("old_name '%s' may miss the prefix '.', old_size=%lu\n",
 				func->old_name, func->old_size);
-		}
 #endif
 
 		if (func->nop)
-- 
2.17.1

